modules/experimental-layers/src/trips-layer/dynamic-trips-layer.js

```diff
@@ -0,0 +1,90 @@
+// Copyright (c) 2015 - 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+import {PathLayer} from '@deck.gl/layers';
+
+const uniformToInject = `\
+uniform float trailLength;
+uniform float currentTime;
+`;
+
+const varyingToInject = `\
+varying float vTime;
+varying float completion;
+varying float xPosition;
+`;
+
+const fragmentShaderToInject = `\
+if(vTime > 1.0 || vTime < 0.0) {
+  discard;
+}
+
+if(xPosition > completion){
+  discard;
+}
+`;
+
+const vertexShaderToInject = `\
+vec4 shift = vec4(0., 0., mod(instanceEndPositions.z, trailLength) * 1e-4, 0.);
+gl_Position = project_to_clipspace(vec4(pos.xy, 1.0, 1.0)) + shift;
```

Comment:
Projection is expensive. Pass in the correct vertex positions so that you don't need to overwrite the logic in the PathLayer.